include/open62541/plugin/pki.h

```diff
@@ -30,6 +30,29 @@ _UA_BEGIN_DECLS
  * The lifecycle of the plugin is attached to a server or client config. The
  * ``clear`` method is called automatically when the config is destroyed. */
 
+/* Enum and functions for the private key password */
+typedef enum  {
+    UA_PRIVATEKEYPASSWORDSTATE_INITIAL = 0,
+    UA_PRIVATEKEYPASSWORDSTATE_WRONG = 1,
+} UA_PrivateKeyPasswordState;
+
+/* This callback is called when a client with security is initialized with a password
+ * protected private key in the PEM format.
+ * If a wrong password is encountered, the callback will be called repeatedly until
+ * doRetry is set to false. */
+typedef UA_String (*UA_PrivateKeyPasswordCallback)(UA_PrivateKeyPasswordState state,
```

Comment:
With the suggested approach, the feature will be completely client specific and any later application to the server or other places where a private key with password may be encountered will require duplicating code.
Are you sure that this will remain restricted to the client?